### PR TITLE
Elasticsearch use testcontainers

### PR DESCRIPTION
**Pull request by rpalcolea**

### CODE REVIEW

File: src/integTest/groovy/nebula/plugin/metrics/ESMetricsPluginIntegTest.groovy [hunk 1]
```diff
@@ -17,60 +17,37 @@
...
+    private final ObjectMapper objectMapper = new ObjectMapper()
 
-        def m2 = output =~ /(Creating|Using) index (.*) for metrics/
-        String index = m2[0][2]
+    @Shared
+    ElasticsearchContainer container = new ElasticsearchContainer("docker.elastic.co/elasticsearch/elasticsearch:5.4.1")
 
-        return [buildId, index]
+    def setup() {
+        while(!container.running) {
```
Comment (bsideup) [inline src/integTest/groovy/nebula/plugin/metrics/ESMetricsPluginIntegTest.groovy:43]: FYI the container either enters `running` state or throws an exception in `.start()` method (called by Spock integration), no need to do this while loop

File: src/integTest/groovy/nebula/plugin/metrics/ESMetricsPluginIntegTest.groovy [hunk 2]
```diff
@@ -17,60 +17,37 @@
...
+import org.testcontainers.spock.Testcontainers
+import spock.lang.IgnoreIf
 import spock.lang.Shared
 
 /**
  * Integration tests for {@link MetricsPlugin}.
  */
+@Slf4j
+@Testcontainers
+@IgnoreIf({ Boolean.valueOf(env["CI"]) }) //TODO: remove once we figure out the stability issues with Travis
```
Comment (bsideup) [inline src/integTest/groovy/nebula/plugin/metrics/ESMetricsPluginIntegTest.groovy:34]: Could you please share the issue with us (Testcontainers)? Maybe we can help solving it :)

Comment (bsideup) [reply src/integTest/groovy/nebula/plugin/metrics/ESMetricsPluginIntegTest.groovy:34]: I checked your Travis config.
It should work with:
```yaml
sudo: required
services:
  - docker
```

**Comment by rpalcolea** [comment]
Hi @bsideup , thanks for looking into this! I wonder how did you find this?

I made the changes https://github.com/nebula-plugins/gradle-metrics-plugin/pull/48 https://github.com/nebula-plugins/gradle-metrics-plugin/pull/49

**Comment by rpalcolea** [comment]
Now we use the elastic testcontainer :) super great

**Comment by dadoonet** [comment]
Amazing!

**Comment by bsideup** [comment]
@rpalcolea
> I wonder how did you find this?

I monitor code on GitHub with "testcontainers" search and was happy to spot this PR =)
We do have "who is using" [section in the docs](https://github.com/testcontainers/testcontainers-java/blob/c8ec853e5095b33dcdeb7a7342243ae4a649f914/docs/index.md#who-is-using-testcontainers) but most of the times usages remain unreported and we do the search to learn about the use cases and potential problems our users might have :)

**Pull request merged**